### ISSUE TITLE
discrepancy between lock content for different Mx processors

### ISSUE THREAD

**Issue by frichtarik** [python-poetry/poetry#9725]
### Description

i'm getting a strange discrepancy when running `poetry lock --no-updates` on mac with M1 processor vs M2/M3 for PyYAML package v6.0.1

for M2/M3, there is one additional file in the lock file

{file = "PyYAML-6.0.1-cp312-cp312-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:a08c6f0fe150303c1c6b71ebcd7213c2858041a7e01975da3a99aed1e7a378ef"}, 

which then causes "pyproject.toml changed significantly since poetry.lock was last generated." on M1 processor macs.

If i do it other way around and generate lock on M1, this line is not present in the lock and lock is accepted by poetry install on all Mx without a hitch

### Workarounds

use M1 to generate lock containing pyyaml package

### Poetry Installation Method

pipx

### Operating System

Sonoma 14.7

### Poetry Version

Poetry (version 1.8.3)

### Poetry Configuration

```bash session
cache-dir = "/Users/filiprichtarik/Library/Caches/pypoetry"
experimental.system-git-client = false
installer.max-workers = null
installer.modern-installation = true
installer.no-binary = null
installer.parallel = true
keyring.enabled = true
repositories.codeartifact.url = "https://<anonymized>.d.codeartifact.eu-west-1.amazonaws.com/pypi/shared/simple/"
solver.lazy-wheel = true
virtualenvs.create = true
virtualenvs.in-project = null
virtualenvs.options.always-copy = false
virtualenvs.options.no-pip = false
virtualenvs.options.no-setuptools = false
virtualenvs.options.system-site-packages = false
virtualenvs.path = "{cache-dir}/virtualenvs"  # /Users/filiprichtarik/Library/Caches/pypoetry/virtualenvs
virtualenvs.prefer-active-python = true
virtualenvs.prompt = "{project_name}-py{python_version}"
warnings.export = false
```

### Poetry Runtime Logs

```bash session
there is no failing command
```

**Comment by Secrus** [comment]
Did you try clearing the cache?

**Comment by frichtarik** [comment]
this is not my own issue - i've updated lock on my M3 and several team members with M1 did encounter  "pyproject.toml changed significantly since poetry.lock was last generated." 

this can be reproduced with clear cache on both sides 

second option is that there is something fishy within pyyaml configuration itself as we didn't encounter it with any other package

**Comment by dimbleby** [comment]
> which then causes "pyproject.toml changed significantly since poetry.lock was last generated." on M1 processor macs.

this is not correct.  The cause of that message is that the local pyproject.toml file has been changed since poetry.lock was generated, and the "fix" is just to regenerate the lock file.

additional files showing up in the lock file _is_ likely a result of caching and can be resolved by clearing the cache, but these are two separate questions

**Comment by frichtarik** [comment]
good morning. 
finally I was able to organize colleagues and collect everything related

i've created a small sample project with single dependency and had people with different Mx processors run poetry lock at it 

files can be found here 

https://github.com/frichtarik/debug-poetry-pyyaml 
`M1` branch contains files produced by M1 macbookpro
`M3` branch contains files produced by M3 macbookpro

feel free to try yourself if such devices are available to you

feel free to ask for more data to help you investigate

the issue is real - M2/M3 produces additional package `{file = "PyYAML-6.0.1-cp312-cp312-manylinux_2_17_aarch64.manylinux2014_aarch64.whl", hash = "sha256:a08c6f0fe150303c1c6b71ebcd7213c2858041a7e01975da3a99aed1e7a378ef"}, `

please notice that it's python 3.12 only, it's not adding the package for any other python version

**Comment by dimbleby** [comment]
Those who are not seeing that file need to clear their cache

**Comment by dimbleby** [comment]
Also https://github.com/frichtarik/debug-poetry-pyyaml/compare/M1..M3 does not show the difference that you claim anyway

**Comment by abn** [comment]
@frichtarik looks like the issue was to do with stale cache. The `--no-cache` option during locking also helps. 

Closing this for now, please let us know if the issue is not cache related and still exists.
